src/reporter.ts

```diff
@@ -162,10 +163,29 @@ export default class SauceReporter extends SummaryFormatter {
       });
       curr.addTest(test);
     });
+    this.calculateTestTiming(curr, endTime);
     suite.addSuite(curr);
     this.testRun.addSuite(suite);
   }
 
+  // Calculate each test's startTime and videoTimestamp.
+  calculateTestTiming(suite: Suite, endTime: Date) {
+    let currEndTime = endTime;
+
+    // Use reduceRight to iterate through the tests from last to first.
+    suite.tests.reduceRight((_, test) => {
```

Comment:
`reduce*` functions aren't meant for iteration. Lots of other, simpler, ways to iterate in reverse order.